lib/util/traffic-switch/traffic-switcher-i-token-fee-fetcher.ts

```diff
@@ -31,19 +31,28 @@ export class TrafficSwitcherITokenFeeFetcher extends TrafficSwitcher<ITokenFeeFe
       return
     }
 
-    // We have results from both implementations, compare them as a whole
-    const identical = JSON.stringify(resultA) === JSON.stringify(resultB)
+    // We have results from both implementations, compare them as a whole.
+    // Before comparison, do some cleaning and keep only entries with a fee != 0.
+    // This is needed as different implementations can return empty/null entry, or entry with 0 fees.
+    const cleanResult = (result: TokenFeeMap): TokenFeeMap =>
```

Comment:
this feels slightly over complicated. if we can just filter out the fee === undefined and fee === 0, then both results are sanitized to compare

Reply:
we can discuss this - i wanted to only make changes in the comparison method first, then we can figure out how to clean results in the online path.